node/core/approval-voting/src/lib.rs

```diff
@@ -768,25 +866,50 @@ async fn handle_actions(
 				let block_hash = indirect_cert.block_hash;
 				let validator_index = indirect_cert.validator;
 
-				ctx.send_unbounded_message(ApprovalDistributionMessage::DistributeAssignment(
-					indirect_cert,
-					candidate_index,
-				).into());
+				match currently_checking_set.approvals_cache.get(&candidate_hash) {
+					Some(ApprovalOutcome::Approved) => {
+						let mut sender = ctx.sender().clone();
+						let _ = issue_approval(
```

Comment:
Is this dropping actions? These need to be processed so we can update the metadata on-disk.

Reply:
This might be vulnerable to #3311 as well.